src/Configuration/Types.hs

```diff
@@ -0,0 +1,383 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
```

Comment:
Maybe drop this and make everything very explicit? No need to have people wonder how the structure got filled with values, and this is not really good if we want to use thing configuration elsewhere. Erik had a nice talk about this.